src/WebGL.ts

```diff
@@ -2199,9 +2208,9 @@ export class WebGLRenderingContext {
 	@inline activeTexture(texture: GLenum): void {
 		activeTexture(this.gl_id, texture);
 	}
-	@inline createContextFromCanvas(canvas_id: string, context_type: string): WebGLRenderingContext {
-		this.gl_id = createContextFromCanvas(canvas_id, context_type);
-		return this.gl_id;
+
+	@inline static createContextFromCanvas(canvas_id: string, context_type: string): WebGLRenderingContext {
+    return new WebGLRenderingContext(canvas_id, context_type)
```

Comment:
This returns a new class with the new id instead of modifying the existing "instance".